examples/graceful_shutdown.py

```diff
@@ -5,46 +5,51 @@
 from contextlib import suppress
 from datetime import datetime
 from functools import partial
+from typing import Any, Callable, Dict, Optional
 
 from aiohttp import web
 
 from aiohttp_sse import EventSourceResponse, sse_response
 
-streams = web.AppKey("streams", weakref.WeakSet[EventSourceResponse])
-worker = web.AppKey("worker", asyncio.Task[None])
+streams_key = web.AppKey("streams_key", weakref.WeakSet["SSEResponse"])
+worker_key = web.AppKey("worker_key", asyncio.Task[None])
 
 
 class SSEResponse(EventSourceResponse):
     async def send_json(
         self,
-        data,
-        id=None,
-        event=None,
-        retry=None,
-        json_dumps=partial(json.dumps, indent=2),
-    ):
+        data: Dict[str, Any],
+        id: Optional[str] = None,
+        event: Optional[str] = None,
+        retry: Optional[int] = None,
+        json_dumps: Callable[..., str] = partial(json.dumps, indent=2),
+    ) -> None:
         await self.send(json_dumps(data), id=id, event=event, retry=retry)
 
 
-async def send_event(stream, data, event_id):
+async def send_event(
+    stream: SSEResponse,
+    data: Dict[str, Any],
+    event_id: str,
+) -> None:
     try:
-        await stream.send_json(data, id=str(event_id))
+        await stream.send_json(data, id=event_id)
     except Exception:
         logging.exception("Exception when sending event: %s", event_id)
 
 
-async def worker(app):
+async def worker(app: web.Application) -> None:
     while True:
         now = datetime.now()
         delay = asyncio.create_task(asyncio.sleep(1))  # Fire
 
         fs = []
-        for stream in app[streams]:
+        for stream in app[streams_key]:
             data = {
                 "time": f"Server Time : {now}",
                 "last_event_id": stream.last_event_id,
             }
-            coro = send_event(stream, data, now.timestamp())
+            coro = send_event(stream, data, str(now.timestamp()))
```

Comment:
Seems like we could just make the id an int and leave this unchanged?

Reply:
`id` was typed as a `str`...
But any value will become `str` after f-string (so float timestamp will be converted well).

But mypy asks to use expected type only:
<img width="680" alt="Screenshot 2024-01-31 at 01 31 11" src="https://github.com/aio-libs/aiohttp-sse/assets/25399456/b0755b2b-6cf1-40dc-9537-382140382d0b">

Should we change expected type from "str" to "Any" or some union?

Reply:
Ah, I misread, I think you're right with str.